mypy/test/data.py

```diff
@@ -561,13 +549,37 @@ def pytest_addoption(parser: Any) -> None:
 def pytest_pycollect_makeitem(collector: Any, name: str, obj: Any) -> Any:
     if not isinstance(obj, type) or not issubclass(obj, DataSuite):
         return None
+    if obj is DataSuite:
+        return None
     return MypyDataSuite(name, parent=collector)
 
 
 class MypyDataSuite(pytest.Class):  # type: ignore  # inheriting from Any
     def collect(self) -> Iterator['MypyDataCase']:
-        for case in self.obj.cases():
-            yield MypyDataCase(case.name, self, case)
+        cls = self.obj
```

Comment:
Add comment -- what is `self.obj` here and where does it come from? This apparently requires understanding some pytest magic, which many readers likely won't be too familiar with, including me.